src/transaction/log_recovery_analysis.cpp

```diff
@@ -554,6 +557,45 @@ log_rv_analysis_undo_redo (THREAD_ENTRY *thread_p, int tran_id, LOG_LSA *log_lsa
   if (is_mvcc)
     {
       tdes->last_mvcc_lsa = *log_lsa;
+
+      // assign transaction mvccid from log record to transaction descriptor
+      assert (log_page_p != nullptr);
+
+      // move read pointer past the log header which is actually read upper in the stack
+      LOG_READ_ADD_ALIGN (thread_p, sizeof (LOG_RECORD_HEADER), log_lsa, log_page_p);
+      switch (log_type)
```

Comment:
Nit : Can this code be templated on log record data struct ?

Reply:
It can be done but it would be more verbose.
However, I separated non-mvcc and mvcc analysis functions.